Sources/SKCore/BuildServerBuildSystem.swift

```diff
@@ -171,13 +167,11 @@ public final class BuildServerBuildSystem: MessageHandler {
   /// the build server has sent us a notification.
   ///
   /// We need to notify the delegate about any updated build settings.
-  public func handle(_ params: some NotificationType, from clientID: ObjectIdentifier) {
-    queue.async {
-      if let params = params as? BuildTargetsChangedNotification {
-        self.handleBuildTargetsChanged(Notification(params, clientID: clientID))
-      } else if let params = params as? FileOptionsChangedNotification {
-        self.handleFileOptionsChanged(Notification(params, clientID: clientID))
-      }
+  public func handle(_ params: some NotificationType, from clientID: ObjectIdentifier) async {
+    if let params = params as? BuildTargetsChangedNotification {
+      await self.handleBuildTargetsChanged(Notification(params, clientID: clientID))
+    } else if let params = params as? FileOptionsChangedNotification {
+      await self.handleFileOptionsChanged(Notification(params, clientID: clientID))
     }
```

Comment:
Same question regarding waiting for the delegate method to complete, looks like conformers of BuildSystemDelegate end up dispatching a `Task` asynchronously, so I don't think it's an issue, just wondering if it's something guaranteed by BuildSystemDelegate (or whether it even matters)

Reply:
Once `BuildSystemDelegate` has been asyncified, the implementers of `BuildSystemDelegate` will no longer create tasks. Again, the reason for awaiting here are the ordering concerns. We can revisit whether ordering is relevant once the functions in `BuildSystemManager` have been re-designed to no longer be order-dependent: https://github.com/ahoppen/sourcekit-lsp/commit/e22d1a4965875391b6b9297137761cce41a5081b